code/data_structures/src/list/doubly_linked_list/doubly-linked-list.c

```diff
@@ -0,0 +1,287 @@
+               return;  
+           }  
+       }  
+       printf("Enter value");  
+       scanf("%d",&item);  
+       ptr->data = item;  
+       ptr->next = temp->next;  
+       ptr -> prev = temp;  
+       temp->next = ptr;  
+       temp->next->prev=ptr;  
+       printf("\nnode inserted\n");  
+   }  
+}  
+void deletion_beginning()  
+{  
+    struct node *ptr;  
+    if(head == NULL)  
+    {  
+        printf("\n UNDERFLOW");  
+    }  
+    else if(head->next == NULL)  
+    {  
+        head = NULL;   
+        free(head);  
+        printf("\nnode deleted\n");  
+    }  
+    else  
+    {  
+        ptr = head;  
+        head = head -> next;  
+        head -> prev = NULL;  
+        free(ptr);  
+        printf("\nnode deleted\n");  
+    }  
+  
+}  
+void deletion_last()  
+{  
+    struct node *ptr;  
+    if(head == NULL)  
+    {  
+        printf("\n UNDERFLOW");  
+    }  
+    else if(head->next == NULL)  
+    {  
+        head = NULL;   
+        free(head);   
+        printf("\nnode deleted\n");  
+    }  
+    else   
+    {  
+        ptr = head;   
+        if(ptr->next != NULL)  
+        {  
+            ptr = ptr -> next;   
+        }  
+        ptr -> prev -> next = NULL;   
+        free(ptr);  
+        printf("\nnode deleted\n");  
+    }  
+}  
+void deletion_specified()  
+{  
+    struct node *ptr, *temp;  
+    int val;  
+    printf("\n Enter the data after which the node is to be deleted : ");  
+    scanf("%d", &val);  
+    ptr = head;  
+    while(ptr -> data != val)  
+    ptr = ptr -> next;  
+    if(ptr -> next == NULL)  
+    {  
+        printf("\nCan't delete\n");  
+    }  
+    else if(ptr -> next -> next == NULL)  
+    {  
+        ptr ->next = NULL;  
+    }  
+    else  
+    {   
+        temp = ptr -> next;  
+        ptr -> next = temp -> next;  
+        temp -> next -> prev = ptr;  
+        free(temp);  
+        printf("\nnode deleted\n");  
+    }     
+}  
+void display()  
+{  
+    struct node *ptr;  
+    printf("\n printing values...\n");  
+    ptr = head;  
+    while(ptr != NULL)  
+    {  
+        printf("%d\n",ptr->data);  
+        ptr=ptr->next;  
+    }  
+}   
+void search()  
+{  
+    struct node *ptr;  
+    int item,i=0,flag;  
+    ptr = head;   
+    if(ptr == NULL)  
+    {  
+        printf("\nEmpty List\n");  
+    }  
+    else  
+    {   
+        printf("\nEnter item which you want to search?\n");   
+        scanf("%d",&item);  
+        while (ptr!=NULL)  
+        {  
+            if(ptr->data == item)  
+            {  
+                printf("\nitem found at location %d ",i+1);  
+                flag=0;  
+                break;  
+            }   
+            else  
+            {  
+                flag=1;  
+            }  
+            i++;  
+            ptr = ptr -> next;  
+        }  
+        if(flag==1)  
+        {  
+            printf("\nItem not found\n");  
+        }  
+    }     
+   }  
+
```

Comment:
@carrisunrio Please remove unwanted line breaks. Also, follow google style guide which will enhance the readability of code.